### PR TITLE
Redesign unit_lib to support measurement aliases.

### PR DESCRIPTION
**Pull request by dplute**
This branch adds the proposed measurement aliasing feature to the ontology validator.

* Added syntax for measurements that copy another measurement.
* Adjusted UnitUniverse and UnitNamespace methods for better encapsulation.
* Removed the Unit.measurement_type field. Instead, UnitNamespace now maintains a map from measurements to the list of units that belong to it.
* At the end of processing the unit file, each aliased measurement copies the units that were defined in the base measurement.
* 'no_units' is still the only unit name that is allowed to appear more than once in the units YAML file, any other duplication is achieved through measurement aliasing.

### CODE REVIEW

File: tools/validators/ontology_validator/yamlformat/validator/unit_lib.py [hunk 1]
```diff
@@ -28,31 +29,49 @@
...
 
     Args:
       namespace: UnitNamespace
 
     Returns:
-      The units map from the namespace.
+      The namespace.
     """
-    return namespace.units
+    return namespace
```
Comment (kswang2400) [inline tools/validators/ontology_validator/yamlformat/validator/unit_lib.py:53]: i'm not seeing where this function is being used, but it seems pretty pointless now? can we just delete this?

File: tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py [hunk 2]
```diff
@@ -170,26 +173,74 @@ def testUnitFolderAddFromConfigTooManyStandardUnits(self):
...
+      'length': 'distance',
+      'level': 'length',
+    }
+    folder = unit_lib.UnitFolder(_GOOD_PATH)
+
+    folder.AddFromConfig([doc], '{0}/file.yaml'.format(_GOOD_PATH))
+
+    self.assertIsInstance(folder.local_namespace.GetFindings()[0],
+                          findings_lib.MeasurementAliasIsAliasedError)
+
```
Comment (kswang2400) [inline tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py:243]: add a test for DuplicateMeasurementAliasError?

```
    doc = {
      'distance': [
        {'meters': 'STANDARD'},
        'feet'
      ],
      'level': 'distance',
    }

    doc2 = {
      'length': [
        {'meters': 'STANDARD'},
        'feet'
      ],
      'level': 'length',
    }
```

**Review by kswang2400 — APPROVED** [review]
minor comments, LGTM

File: tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py [hunk 3]
```diff
@@ -170,26 +173,74 @@ def testUnitFolderAddFromConfigTooManyStandardUnits(self):
...
+    }
+    folder = unit_lib.UnitFolder(_GOOD_PATH)
+    expected_units = ['meters', 'feet']
+
+    folder.AddFromConfig([doc], '{0}/file.yaml'.format(_GOOD_PATH))
+    distance_units = folder.local_namespace.GetUnitsForMeasurement('distance')
+    length_units = folder.local_namespace.GetUnitsForMeasurement('length')
+
+    self.assertEmpty(folder.local_namespace.GetFindings())
+    self.assertCountEqual(expected_units, distance_units)
```
[the remark this answers is not in the document]
Comment (dplute) [reply tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py:210]: assertCountEqual actually does check that both sequences contain the elements, but they are not required to be in the same order.  It's just a weird name for what the method does.

File: tools/validators/ontology_validator/yamlformat/validator/presubmit_validate_types_lib.py [hunk 4]
```diff
@@ -171,9 +156,6 @@ def GetUnitsForMeasurement(self,
         incremented.
     Returns: a string representing the unit or None if units are defined.
     """
-    if not self.unit_universe_reverse_map:
```
[the remark this answers is not in the document]
Comment (dplute) [reply tools/validators/ontology_validator/yamlformat/validator/presubmit_validate_types_lib.py:174]: Added a check that returns if self.unit_universe is empty.

File: tools/validators/ontology_validator/yamlformat/validator/unit_lib.py (hunk 1 above)
Comment (dplute) [reply tools/validators/ontology_validator/yamlformat/validator/unit_lib.py:53]: This is part of the way FindingsUniverse works and it's populating the _namespace_map that the other methods use.

File: tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py (hunk 2 above)
Comment (dplute) [reply tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py:243]: Added

File: tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kswang2400) [reply tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py:210]: got it, thanks for clarifying :)

File: tools/validators/ontology_validator/yamlformat/validator/unit_lib.py (hunk 1 above)
Comment (kswang2400) [reply tools/validators/ontology_validator/yamlformat/validator/unit_lib.py:53]: ok, makes sense to leave it for now

**Review by kswang2400 — APPROVED** [review]

**Pull request merged**